### PR TITLE
Convert type any to type unknown

### LINKED ISSUE #4490: no-explicit-any warnings - prefer unknown
> It seems we have delayed after-effects after this merge. #4284 https://flowcrypt.semaphoreci.com/jobs/47dbcb8e-5bc4-437a-9812-e70699e7d855

They are all warnings, except for 

```
   148:4   error    Missing semicolon                         @typescript-eslint/semi 00:09 
```

In either case, we should try to address these warnings - I'll file an issue.

_Originally posted by @tomholub in https://github.com/FlowCrypt/flowcrypt-browser/issues/4484#issuecomment-1152617605_

### PR DESCRIPTION
**Pull request by martgil**
This PR's goal is to convert type any to type unknown. 

To check the warnings on master branch:
`npm run test_eslint`

close #4490 // if this PR closes an issue

issue #0000 // if it doesn't close the issue yet

----------------------------------

**Tests** _(delete all except exactly one)_:
- Does not need tests (refactor only, docs or internal changes)
- Difficult to test (explain why)
- Not worth testing
- Tests will be added later (issue #...)
- Tests added or updated

--------------------------------

### To be filled by reviewers

I have reviewed that this PR... _(tick whichever items you personally focused on during this review)_:
- [ ] addresses the issue it closes (if any)
- [ ] code is readable and understandable
- [ ] is accompanied with tests, or tests are not needed
- [ ] is free of vulnerabilities
- [ ] is documented clearly and usefully, or doesn't need documentation

### CODE REVIEW

**Comment by martgil** [comment]
Hello sir, @ioanmo226 - I initially started with converting 5 of these... there is some type of checking in common places like catching an error... if this looks okay maybe I can start converting type any to unknown from there?

File: extension/chrome/elements/compose-modules/compose-send-btn-module.ts [hunk 1]
```diff
@@ -249,9 +249,9 @@ export class ComposeSendBtnModule extends ViewModule<ComposeView> {
   private doSendMsgs = async (msgObj: MultipleMessages): Promise<SendMsgsResult> => {
     const sentIds: string[] = [];
     const supplementaryOperations: Promise<void>[] = [];
-    const supplementaryOperationsErrors: any[] = []; // tslint:disable-line:no-unsafe-any
+    const supplementaryOperationsErrors: unknown[] = []; // tslint:disable-line:no-unsafe-any
```
Comment (ioanmo226) [inline extension/chrome/elements/compose-modules/compose-send-btn-module.ts:252]: you may remove `// tslint:disable-line:no-unsafe-any` comment if you already converted `any` to `unknown`
There are several places which contains this comment

Comment (martgil) [reply extension/chrome/elements/compose-modules/compose-send-btn-module.ts:252]: oh - ok, thanks... I'll be removing them as well. thank you!

File: extension/types/squire.d.ts [hunk 2]
```diff
@@ -1,9 +1,12 @@
...
+
 export as namespace SquireClass;
 
 declare global {
   interface Window {
     // @ts-ignore
     Squire: SquireClass;
+    forge: typeof forge;
   }
 }
```
Comment (tomholub) [inline extension/types/squire.d.ts:12]: Also this is wrong to add forge into squire definitions. Let's remove it and file an issue for Ioan to have a look later.

Comment (martgil) [reply extension/types/squire.d.ts]: okay sir, I'll remove it - I just converted most of it from any to unknown but there are still some that need to have proper types but I can't convert them just yet.

Comment (tomholub) [reply extension/types/squire.d.ts]: That's ok. You can leave these for another PR in the future.

File: extension/js/common/core/const.ts [hunk 3]
```diff
@@ -3,7 +3,7 @@
 'use strict';
 
 export const VERSION = '[BUILD_REPLACEABLE_VERSION]';
-export const FLAVOR: 'consumer' | 'enterprise' = '[BUILD_REPLACEABLE_FLAVOR]' as any;
+export const FLAVOR: 'consumer' | 'enterprise' = '[BUILD_REPLACEABLE_FLAVOR]' as unknown as 'consumer' | 'enterprise';
```
Comment (ioanmo226) [inline extension/js/common/core/const.ts:6]: I think we don't need `as unknown` here.
`as 'consumer' | 'enterprise'` will be enough

Comment (tomholub) [reply extension/js/common/core/const.ts:6]: It wouldn't work because `'[BUILD_REPLACEABLE_FLAVOR]'` is not of that type. It's a bit of a special case.

Comment (ioanmo226) [reply extension/js/common/core/const.ts:6]: It works...
Tslint and eslint doesn't return any error and vscode displays fine result.

image

Comment (ioanmo226) [reply extension/js/common/core/const.ts:6]: This way works because left type is string union type and right type is string.

Comment (tomholub) [reply extension/js/common/core/const.ts:6]: Ah - good. Can be done separately. Thanks for noticing.

**Pull request merged**